app/assets/javascripts/components/tasks/TaskForm.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import AsyncForm from 'components/forms/AsyncForm';
+
+class TaskForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this._getFormData = this._getFormData.bind(this);
+  }
+
+  _getFormData() {
+    return {
+      task: this.props.task,
+      taskName: this.input.value
+    };
+  }
+
+  render() {
+    return (
+      <AsyncForm
+        getFormData={this._getFormData}
+        onSubmit={this.props.onSubmit}
+      >
+        {
+          ({loading, handleSubmit}) => (
+            <div className='task_form clearfix'>
+              <input
+                name='task'
+                defaultValue=''
+                disabled={loading}
+                className='form-control input-sm input-task'
+                ref={(input) => { this.input = input }}
```

Comment:
We can create this function on the constructor too and avoid to recreate it on every re-render

```js
constructor() {
  this.saveInputRef = (input) => { this.input = input };
}
// ...
  ref={this.saveInputRef}
// ...
```

You can do it for the other places you save refs too.